dashboard/app/controllers/api/v1/section_instructors_controller.rb

```diff
@@ -0,0 +1,95 @@
+require 'cdo/firehose'
+
+class Api::V1::SectionInstructorsController < Api::V1::JSONApiController
+  # Returns list of current user's SectionInstructor records
+  # GET /section_instructors
+  def index
+    section_instructors = SectionInstructor.where(instructor_id: current_user&.id)
+
+    render json: section_instructors, each_serializer: Api::V1::SectionInstructorSerializer
+  end
+
+  # Returns SectionInstructor records for the given section
+  # GET /section_instructors/:section_id
+  def show
+    section = Section.find(params.require(:section_id))
+    authorize! :manage, section
+    section_instructors = SectionInstructor.where(section: section)
+
+    render json: section_instructors, each_serializer: Api::V1::SectionInstructorSerializer
+  end
+
+  # Adds a new teacher as an invited instructor
+  # POST /section_instructors
+  def create
+    section = Section.find(params.require(:section_id))
+    authorize! :manage, section
+
+    # Enforce maximum co-instructor count (the limit is 5 plus the main teacher
+    # for a total of 6)
+    if SectionInstructor.where(section: section).count >= 6
+      return render json: {error: 'section full'}, status: :bad_request
+    end
+
+    instructor = User.find_by(email: params.require(:email), user_type: :teacher)
+    return head :not_found if instructor.blank?
+
+    si = SectionInstructor.with_deleted.find_by(instructor: instructor, section: section)
+
+    # Actually delete the instructor if they were soft-deleted so they can be re-invited.
+    if si&.deleted_at.present?
+      si.really_destroy!
+    # Can't re-add someone who is already an instructor (or invited/declined)
+    elsif si.present?
+      return render json: {error: 'already invited'}, status: :bad_request
+    end
+
+    si = SectionInstructor.create!(
+      section: section,
+      instructor: instructor,
+      status: :invited,
+      invited_by: current_user
+    )
+
+    render json: si, serializer: Api::V1::SectionInstructorSerializer
+  end
+
+  # Removes an instructor from the section (soft-deleting the record).
+  # DELETE /section_instructors/:id
+  def destroy
+    si = SectionInstructor.find(params.require(:id))
+    authorize! :manage, si.section
```

Comment:
curious if it's worth using load_and_authorize_resource here? (from CanCanCan)

Reply:
I just made that change. I don't think it made a significant difference in code legibility, but working through it did help me better understand the CanCanCan library.